examples/jsm/exporters/USDZExporter.js

```diff
@@ -617,53 +1239,330 @@ function buildVector2( vector ) {
 
 }
 
+class RegisteredAnimationInfo {
+
+	get start() {
+
+		return this.ext.getStartTime01( this.root, this.clip );
+
+	}
+	get duration() {
+
+		return this.clip.duration;
+
+	}
+
+	constructor( ext, root, clip ) {
+
+		this.ext = ext;
+		this.root = root;
+		this.clip = clip;
+
+	}
+
+}
+
+class TransformData {
+
+	get frameRate() {
+
+		return 60;
+
+	}
+
+	constructor( ext, root, target, clip ) {
+
+		this.ext = ext;
+		this.root = root;
+		this.target = target;
+		this.clip = clip;
+
+	}
+
+	addTrack( track ) {
+
+		if ( track.name.endsWith( 'position' ) )
+			this.pos = track;
+
+		if ( track.name.endsWith( 'quaternion' ) )
+			this.rot = track;
+
+		if ( track.name.endsWith( 'scale' ) )
+			this.scale = track;
+
+	}
+
+	getFrames() {
+
+		var _a, _b, _c, _d, _e, _f, _g, _h, _j;
+		return Math.max( ( _c = ( _b = ( _a = this.pos ) === null || _a === void 0 ? void 0 : _a.times ) === null || _b === void 0 ? void 0 : _b.length ) !== null && _c !== void 0 ? _c : 0, ( _f = ( _e = ( _d = this.rot ) === null || _d === void 0 ? void 0 : _d.times ) === null || _e === void 0 ? void 0 : _e.length ) !== null && _f !== void 0 ? _f : 0, ( _j = ( _h = ( _g = this.scale ) === null || _g === void 0 ? void 0 : _g.times ) === null || _h === void 0 ? void 0 : _h.length ) !== null && _j !== void 0 ? _j : 0 );
```

Comment:
What is this?

Reply:
> (Note: the Animation-related code is converted from TypeScript, there may still be a few traces of strange generated syntax left)

I'm happy to clean this up if there's interest.